### ISSUE TITLE
Model a lot slower and larger after quantization

### ISSUE THREAD

**Issue by urbste** [clevercool/SQuant#2]
Hi, thanks for releasing the source code to your paper and your great work :-)

I just tried out your code. Quantization works, however, when I test the model it is a lot slower than the FP32 version:

I execute it like this:
```bash
python main.py --batch_size 256 -wb 8 -ab 8
```

And I added some code at the end of main.py:
```python
import time
def measure_time(model, rand_i, runs = 100):
    for i in range(10):
        model(rand_input)
    start = time.time()
    for i in range(runs):
        model(rand_input)
    print("Runtime {:.2f}".format((time.time()-start)/runs))
    
with torch.no_grad():
    model = model.cuda()
    model.eval()
    measure_time(model, rand_input,10)
    quantized_model.eval()
    measure_time(quantized_model, rand_input,10)

    torch.save(model, "original.ckpt")
    torch.save(quantized_model, "quantized.ckpt")
```
Output:
```bash
Average runtime 0.16s
Average runtime 0.27s
```
So the quantized model (second timing) is a lot slower. In addition, when I save the quantized model it has almost twice the size.

What am I doing wrong? 
Is there something we need to do after your quantization algorithm like freezing the model somehow? Could you maybe add an explanation to your repo how to fully harness the quantized model (like having faster inference speed and smaller model size)?
I am running the code on a GTX 1080 with CUDA 11 and Pytorch 1.11.

**Comment by clevercool** [comment]
Hi, Thanks for your attention to our work.
Our implementation is fake quantization, which simulates the Int8 using FP32, following other quantization work.
For a fair comparison, our quantization module code is based on ZeroQ. 
Also, the baselines (including ZeroQ, DSG) that we adopt in our paper are all based on fake quantization.
You can refer to this [issue](https://github.com/amirgholami/ZeroQ/issues/2).

Therefore, the model size won't be smaller, and the speed will be slower due to the extra fake quantization function for activations.

**Comment by urbste** [comment]
Thanks for the clarification. I just started reading into the quantization topic and your code looked super easy to reproduce. 
But for practical applications, how would I convert the model for Int8 inference on, e.g. an embedded device?

I looked into [this](https://pytorch.org/docs/stable/quantization.html), however, simply using this interface did not work ;-)

I am also linking to the issue in the ZeroQ repo https://github.com/amirgholami/ZeroQ/issues/16
